### PR TITLE
Upgrade module to PHP7

### PR DESCRIPTION
**Pull request by madmatt**
## API Changes:
- Update to use the Chillu\FakeDatabase namespace for classes
- Rename FakeObject::create_from_array() to FakeObject::createFromArray() (PSR-2 requirement for static methods)
- Update to require PHP >7.2 (could be relaxed, mainly requires this due to phpunit/phpunit:^8)
- Remove unused FakeDatabaseException class

## Test changes:
- Update to phpunit 8 (from 3.7)
- Update code coverage from ~75% to 100%
- Add PHP Code Sniffer and run across codebase

## Misc changes:
- Add .gitignore to keep dev requirements out of the module (including vendor)
- Remove vendor folder and composer.lock from module

### CODE REVIEW

**Comment by madmatt** [comment]
Alright there we go, now Travis builds should pass on PHP7.2 as well (apparently Travis installs but doesn't automatically run `composer install`... who knew?)

File: src/FakeObject.php [hunk 1]
```diff
@@ -1,69 +1,80 @@
...
-	 * @return FakeObject
-	 */
-	public static function create_from_array($array) {
-		// array_walk_recursive doesn't recurse into arrays...
-		foreach($array as &$v) {
-			// Convert "has one" relations
-			if(is_array($v)) {
-				$v = FakeObject::create_from_array($v);
-			} 
-			// Convert "has many" relations
```
Comment (chillu) [inline src/FakeObject.php:53]: lol so the `elseif` is the same condition as the `if`. There's no test coverage for `has_many` or `many_many`. Is this a feature regression?

File: composer.json [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
 			"name" : "Ingo Schommer",
 			"email" : "me@chillu.com",
 			"role" : "Developer"
+		},
+		{
+			"name": "madmatt",
+			"email": "signups+github@madman.net.nz"
 		}
 	],
+	"require": {
```
Comment (chillu) [inline composer.json:17]: Wanna set a branch-alias: 2.x as well? So nobody pulls in those breaking changes automatically

**Review by chillu — CHANGES_REQUESTED** [review]

File: src/FakeObject.php (hunk 1 above)
Comment (madmatt) [reply src/FakeObject.php:53]: Yeah I couldn't work out (nor remember) what this was supposed to be for anyway. As far as I could figure from the tests everything was covered by either `not array` or `is array` and I couldn't work out what the comments 'has one' and 'has many' implied (except for the distinction between 'it's a single value' and 'it's an array of values' to create)...

Can you remember what this was supposed to be used for?

Comment (chillu) [reply src/FakeObject.php:53]: Nope, if the tests pass, and your use case for the project you're migrating is still satisfied, I'm happy to leave it as-is

Comment (madmatt) [reply src/FakeObject.php]: Yeah I couldn't work out (nor remember) what this was supposed to be for anyway. As far as I could figure from the tests everything was covered by either `not array` or `is array` and I couldn't work out what the comments 'has one' and 'has many' implied (except for the distinction between 'it's a single value' and 'it's an array of values' to create)...

Can you remember what this was supposed to be used for?

**Comment by madmatt** [comment]
Hey @chillu, I've added the branch-alias now. I _think_ I did it right, but can you check? I'm not entirely sure I understand how branch aliases work, so I'm pretty sure I've got it around the right way, but it's like trying to get the argument order to `tar` or `ln -s` the right way around, I never get it right the first time :D

File: composer.json (hunk 2 above)
Comment (madmatt) [reply composer.json:17]: Done :)

File: src/FakeObject.php (hunk 1 above)
Comment (madmatt) [reply src/FakeObject.php:53]: Cool - yeah all tests pass and I can't figure out what it's used for. If I ever do, I can re-instate something :)

**Review by chillu — APPROVED** [review]

**Pull request merged**